spec/lib/pliny/librato/metrics/backend_spec.rb

```diff
@@ -1,49 +1,54 @@
 require 'spec_helper'
 
 RSpec.describe Pliny::Librato::Metrics::Backend do
-  let(:source)   { 'myapp.production' }
-  let(:interval) { 1 }
-  let(:count)    { 5 }
-  let(:queue)    { double('queue') }
-  let(:metrics)  { { 'foo.bar' => 1, baz: 2 } }
+  let(:source)        { 'myapp.production' }
+  let(:interval)      { 1 }
+  let(:count)         { 5 }
+  let(:librato_queue) { instance_double(Librato::Metrics::Queue) }
+  let(:metrics)       { { 'foo.bar' => 1, baz: 2 } }
 
   subject(:backend) do
     described_class.new(
       count:    count,
       interval: interval,
-      source:   source,
-      queue:    queue
+      source:   source
     )
   end
 
+
   describe '#initialize' do
-    context 'without a provided queue' do
-      let(:queue) { nil }
+    it 'creates a Librato::Metrics::Queue' do
+      expect(Librato::Metrics::Queue).to receive(:new).with(
+        autosubmit_count:    count,
+        autosubmit_interval: interval,
+        source:              source
+      ).and_call_original
+
+      expect(backend.send(:librato_queue))
+        .to be_an_instance_of(Librato::Metrics::Queue)
+    end
 
-      it 'creates a Librato::Metrics::Queue' do
-        expect(Librato::Metrics::Queue).to receive(:new).with(
-          autosubmit_count:    count,
-          autosubmit_interval: interval,
-          source:              source
-        ).and_call_original
+    it 'creates a new Queue' do
+      expect(Queue).to receive(:new).and_call_original
 
-        expect(backend.queue).to be_an_instance_of(Librato::Metrics::Queue)
-      end
+      expect(backend.send(:metrics_queue)).to be_an_instance_of(Queue)
     end
   end
 
   shared_examples 'a metrics reporter' do
-    it 'delegates to queue.add' do
-      expect(queue).to receive(:add).with(metrics)
-      backend.send(method, metrics)
+    before do
+      allow(librato_queue).to receive(:submit)
+      allow(librato_queue).to receive(:add)
+      allow(backend).to receive(:librato_queue).and_return(librato_queue)
```

Comment:
We're stubbing out our own method here; I think we should try to avoid that. This was the reason I added `:librato_queue` to the initializer, you don't have to stub internal methods or `Librato::Metrics::Queue.new`.